x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -1625,30 +1615,20 @@ export class RulesClient {
       await this.unsecuredSavedObjectsClient.update(
         'alert',
         id,
-        this.updateMeta({
+        {
           ...attributes,
+          ...(!attributes.apiKeyOwner && { apiKeyOwner: null }),
```

Comment:
do we need to do _anything_ with these properties on disable?  Seems like we don't, so can't we just remove these two lines?

Reply:
I removed it first but one of the integration tests failed. IDK why it's:
`alerting api integration security and spaces enabled  Alerts - Group 1 alerts disable superuser at space1 should still be able to disable alert when AAD is broken` 

Probably due to the encryption... I think Kibana generates a new encryption key on every update no matter if apiKey changed or not. And if we don't send an apiKey, next time we want to decrypt it it fails...

Reply:
>Probably due to the encryption... I think Kibana generates a new encryption key on every update no matter if apiKey changed or not. And if we don't send an apiKey, next time we want to decrypt the APIKey, it fails...

I can see how the test is still failing. I guess the main purpose is to ensure the disable API call returns successfully which it does, however I can see the `checkAAD` call fail if we don't nullify the API key / API key owner. I think it would be best to remove this check from the test rather than removing the associated API key for scenarios the system administrator didn't properly rotate encryption keys and plan to do so at a later time (which will fix the rule at the same time). Thoughts?

Reply:
Doesn't that mean removing the whole test case?
That case is to test that a rule can be disabled even AAD is broken. if we remove `checkAAD`, case would be testing nothing i think?

Reply:
The test checks two things:
- Ensure when the AAD is broken on a rule, we can still call the API to perform the action (primary reason for the test)
- After an action is performed, test to make sure the internal code didn't break the AAD itself.

In this case, checking AAD after is not as relevant because the it was broken before and remains broken. So the `checkAAD` code can be removed, while the test still ensures we can call the API when the AAD was already broken..

The `checkAAD` was added in a bunch of places with this PR https://github.com/elastic/kibana/pull/53333.

Reply:
Got it! thanks.